testing/sdk_test_upgrade.py

```diff
@@ -64,10 +64,29 @@ def upgrade_downgrade(package_name, running_task_count):
     upgrade_or_downgrade(package_name, running_task_count)
 
 
-def upgrade_or_downgrade(package_name, running_task_count):
+# In the soak cluster, we assume that the Universe version of the framework is already installed.
+# Also, we assume that the Universe is the default repo and the stub repos are already in place,
+# so we don't need to add or remove any repos.
+#
+# (1) Upgrades to test version of framework.
+# (2) Downgrades to Universe version.
+def soak_upgrade_downgrade(package_name, running_task_count, install_options={}):
+    print('Upgrading to test version')
+    upgrade_or_downgrade(package_name, running_task_count, install_options, 'stub-universe')
+
+    print('Downgrading to master version')
+    upgrade_or_downgrade(package_name, running_task_count, install_options)
+
+
+def upgrade_or_downgrade(package_name, running_task_count, install_options={}, package_version=None):
     task_ids = tasks.get_task_ids(package_name, '')
     marathon.destroy_app(package_name)
-    install.install(package_name, running_task_count, check_suppression=False)
+    install.install(
+        package_name,
+        running_task_count,
+        additional_options=install_options,
+        package_version=package_version,
```

Comment:
In the soak cluster, the default universe is at index=0, so it would look there first. I tried to explain that in the comments above, but I could make it clearer.